src/Controllers/Storage/ProcessController.cs

```diff
@@ -187,6 +191,74 @@ public async Task<ActionResult<ProcessHistoryList>> GetProcessHistory(
             return Ok(processHistoryList);
         }
 
+        /// <summary>
+        /// Attempts to acquire a process lock for an instance.
+        /// </summary>
+        /// <param name="instanceOwnerPartyId">The party id of the instance owner.</param>
+        /// <param name="instanceGuid">The id of the instance to lock.</param>
+        /// <param name="request">The lock request containing expiration time.</param>
+        /// <param name="cancellationToken">CancellationToken</param>
+        /// <returns>The lock response with lock key if successful, or Conflict if lock is already held.</returns>
+        [Authorize]
+        [HttpPost("lock")]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [Produces("application/json")]
+        public async Task<ActionResult<ProcessLockResponse>> AcquireProcessLock(
+            int instanceOwnerPartyId,
+            Guid instanceGuid,
+            [FromBody] ProcessLockRequest request,
+            CancellationToken cancellationToken)
+        {
+            var instance = await _instanceRepository.GetOne(instanceOwnerPartyId, instanceGuid);
+
+            if (instance is null)
+            {
+                return NotFound();
+            }
+
+            var lockId = await _processLockRepository.TryAcquireLock(instanceGuid, request.TtlSeconds, cancellationToken);
+
+            if (lockId is null)
+            {
+                return Conflict();
+            }
+
+            return Ok(new ProcessLockResponse { LockId = lockId.Value });
+        }
+
+        /// <summary>
+        /// Updates TTL on a process lock.
+        /// </summary>
+        /// <param name="instanceGuid">The id of the instance to unlock.</param>
+        /// <param name="lockId">The ID of the lock to release.</param>
+        /// <param name="request">The lock request (TTL should be 0 for release).</param>
+        /// <param name="cancellationToken">CancellationToken</param>
+        /// <returns>NoContent if successful.</returns>
+        [Authorize]
+        [HttpPatch("lock/{lockId}")]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Produces("application/json")]
+        public async Task<ActionResult> UpdateProcessLock(
+            Guid instanceGuid,
+            Guid lockId,
+            [FromBody] ProcessLockRequest request,
+            CancellationToken cancellationToken)
+        {
+            bool success = await _processLockRepository.UpdateLockExpiration(instanceGuid, lockId, request.TtlSeconds, cancellationToken);
+
+            if (!success)
+            {
+                return NotFound();
```

Comment:
What could go wrong here, is it he lock ID precondition? Maybe a conflict status?